src/widgetApp/mainwindow.cpp

```diff
@@ -1,16 +1,311 @@
+    routerButton->setMinimumSize(QSize(50,50));
+    routerButton->setMaximumSize(QSize(50,50));
+    graphNetworkFrame->layout()->addWidget(routerButton);
+
+    subnetsGraphFrame->setStyleSheet("background-color: rgb(227, 229, 115, 100)");
+
+    if(isVertical)
+        subnetsGraphFrame->setLayout(new QVBoxLayout());
+    else  subnetsGraphFrame->setLayout(new QHBoxLayout());
+
+    subnetsGraphFrame->layout()->setSpacing(50);
+    graphNetworkFrame->layout()->addWidget(subnetsGraphFrame);
+
+    for(int i = 0; i < subnetCount; i++)
+    {
+         QVBoxLayout *frameLayout = new QVBoxLayout();
+
+         QFrame *subnetFrame = new QFrame();
+         subnetFrame->setLayout(frameLayout);
+
+         subnetsGraphFrame->layout()->addWidget(subnetFrame);
+
+         QString subButtonText = QString::fromStdString("S" + std::to_string(1 + i));
+         QPushButton *subnetButton = new QPushButton(subButtonText);
+
+         subnetButton->setStyleSheet("background-color: rgb(227, 229, 115)");
+         subnetButton->setMinimumSize(QSize(40,40));
+         subnetButton->setMaximumSize(QSize(40,40));
+         subnetFrame->layout()->addWidget(subnetButton);
+
+         QHBoxLayout *hostsLayout;
+
+         for(int j = 0; j < subnets[i]->HostNumber; j++)
+         {
+            if((j + 8) % 8 == 0)
+            {
+                hostsLayout = new QHBoxLayout();
+                QFrame *hostFrame = new QFrame();
+                frameLayout -> addWidget(hostFrame);
+
+                hostFrame->setLayout(hostsLayout);
+                hostFrame->setStyleSheet("background-color: rgb(15, 159, 116,100)");
+                hostFrame->setMaximumWidth(300);
+            }
+
+            QString hostButtonText = QString::fromStdString("H" + std::to_string(1 + j));
+            QPushButton *hostButton = new QPushButton(hostButtonText);
+            hostButton->setStyleSheet("background-color: rgb(15, 159, 116)");
+            hostButton->setMinimumSize(QSize(20,20));
+            hostButton->setMaximumSize(QSize(20,20));
+            hostsLayout->addWidget(hostButton);
+         }
+    }
+}
+
+void widgetApp::MainWindow::displayNetworkInfo()
+{
+    deleteLayoutContent(infoScrollContent);
+
+    for(int i = 0; i < subnetCount; i++)
+    {
+        QVBoxLayout *frameLayout = new QVBoxLayout();
+        QFrame *frame = new QFrame();
+        //frame->setStyleSheet("background-color: rgb(15, 159, 116,100)");
+        infoPanelLayout->addWidget(frame);
+        frame->setMinimumHeight(150);
+        frame->setLayout(frameLayout);
+
+        QString labelText = QString::fromStdString("Subnet " + std::to_string(1 + i));
+        QLabel *subnetLabel = new QLabel(labelText);
+        subnetLabel->font().bold();
+        frame->layout()->addWidget(subnetLabel);
+
+        QLabel *hostLabel = new QLabel(QString::fromStdString(
+                                      "Host number: " + std::to_string((int)subnets.at(i)->HostNumber)));
```

Comment:
Preferowane używanie static_cast<>(), zamiast C-style casting - [cppref](https://en.cppreference.com/w/cpp/language/static_cast)
```suggestion
        QLabel *hostLabel = new QLabel(QString::fromStdString(
                                      "Host number: " + std::to_string(static_cast<int>(subnets.at(i)->HostNumber))));
```